objects/DirectorTimeline.js

```diff
@@ -16,6 +18,31 @@ class DirectorTimeline {
     getDirectorShots() {
         return this.directorShots;
     }
+
+    static fromXML(XMLObject) {
+        const directorTimeline = new DirectorTimeline(
+            XMLObject.description[0]);
+
+        if (XMLObject.shotList[0]) {
+            XMLObject.shotList[0].shot.forEach(shot => {
+                const directorShot = DirectorShot.fromXML(shot);
+                directorTimeline.addDirectorShot(directorShot);
+            });
+        }
+        return directorTimeline;
+    }
+
+    toXML() {
+        const directorShotsXML = [];
+        this.directorShots.forEach((shot) => {
+            directorShotsXML.push(shot.toXML());
+        });
```

Comment:
This can be replaced with a `.map` functional call, e.g. `const directorShotsXML = this.directorShots.map(shot => { return shot.toXML(); });` (Properly indented of course)